### ISSUE TITLE
Completing a hard mode challenge in chapter 6 advances me to chapter 7

### ISSUE THREAD

**Issue by satsie** [saving-satoshi/saving-satoshi#1071]
I'm not sure if this is a bug, but I think it has something to do with state changes that might be off so I'll describe what I'm trying to do here.

I've already played the game up through chapter 8, but now I'm going through all the challenges again to gather a list of curriculum topics for Art. I replayed all of chapter 6 and at the end it advanced me to chapter 7. Then I went back to the list of chapters and toggled hard mode on for chapter 6 because I have not seen how the material is different from normal mode. I'm able to get to https://savingsatoshi.com/en/chapters/chapter-6/in-out-4-hard , which I believe might be the first hard mode challenge in this chapter. I finish the code, submit it, see the congratulations message, and click continue. Instead of advancing me to in-out-5, I am sent straight to the start of chapter 7.

If I check the response for the /v1/progress call, it looks like I did not get credit for completing in-out-4-hard:

```
            {
                "id": 6,
                "completed": true,
                "difficulties": [
                    {
                        "level": "NORMAL",
                        "lessons": [
                            {
                                "id": "CH6INT1",
                                "path": "/chapter-6/intro-1",
                                "completed": true
                            },
                            {
                                "id": "CH6INT2",
                                "path": "/chapter-6/intro-2",
                                "completed": true
                            },
                            {
                                "id": "CH6INO1",
                                "path": "/chapter-6/in-out-1",
                                "completed": true
                            },
                            {
                                "id": "CH6INO2",
                                "path": "/chapter-6/in-out-2",
                                "completed": true
                            },
                            {
                                "id": "CH6INO3",
                                "path": "/chapter-6/in-out-3",
                                "completed": true
                            },
                            {
                                "id": "CH6INO4_NORMAL",
                                "path": "/chapter-6/in-out-4-normal",
                                "completed": true
                            },
                            {
                                "id": "CH6INO5",
                                "path": "/chapter-6/in-out-5",
                                "completed": true
                            },
                            {
                                "id": "CH6PUT1_NORMAL",
                                "path": "/chapter-6/put-it-together-1-normal",
                                "completed": true
                            },
                            {
                                "id": "CH6PUT2_NORMAL",
                                "path": "/chapter-6/put-it-together-2-normal",
                                "completed": true
                            },
                            {
                                "id": "CH6PUT3_NORMAL",
                                "path": "/chapter-6/put-it-together-3-normal",
                                "completed": true
                            },
                            {
                                "id": "CH6OUT1",
                                "path": "/chapter-6/outro-1",
                                "completed": true
                            }
                        ],
                        "completed": true
                    },
                    {
                        "level": "HARD",
                        "lessons": [
                            {
                                "id": "CH6INT1",
                                "path": "/chapter-6/intro-1",
                                "completed": true
                            },
                            {
                                "id": "CH6INT2",
                                "path": "/chapter-6/intro-2",
                                "completed": true
                            },
                            {
                                "id": "CH6INO1",
                                "path": "/chapter-6/in-out-1",
                                "completed": true
                            },
                            {
                                "id": "CH6INO2",
                                "path": "/chapter-6/in-out-2",
                                "completed": true
                            },
                            {
                                "id": "CH6INO3",
                                "path": "/chapter-6/in-out-3",
                                "completed": true
                            },
                            {
                                "id": "CH6INO4_HARD",
                                "path": "/chapter-6/in-out-4-hard",
                                "completed": false
                            },
                            {
                                "id": "CH6INO5",
                                "path": "/chapter-6/in-out-5",
                                "completed": false
                            },
                            {
                                "id": "CH6PUT1_HARD",
                                "path": "/chapter-6/put-it-together-1-hard",
                                "completed": false
                            },
                            {
                                "id": "CH6PUT2_HARD",
                                "path": "/chapter-6/put-it-together-2-hard",
                                "completed": false
                            },
                            {
                                "id": "CH6PUT3_HARD",
                                "path": "/chapter-6/put-it-together-3-hard",
                                "completed": false
                            },
                            {
                                "id": "CH6PUT4_HARD",
                                "path": "/chapter-6/put-it-together-4-hard",
                                "completed": false
                            },
                            {
                                "id": "CH6PUT5_HARD",
                                "path": "/chapter-6/put-it-together-5-hard",
                                "completed": false
                            },
                            {
                                "id": "CH6PUT6_HARD",
                                "path": "/chapter-6/put-it-together-6-hard",
                                "completed": false
                            },
                            {
                                "id": "CH6OUT1",
                                "path": "/chapter-6/outro-1",
                                "completed": false
                            }
                        ],
                        "completed": false
                    }
                ],
                "hasDifficulty": true,
                "selectedDifficulty": "HARD"
            },
```

**Comment by benalleng** [comment]
It looks like we have a regression again whereby completing challenges sends us to the tip of our progress instead of to the next lesson and only pushing progress forward if we are at the progress tip.

If you keep completing the same page over and over again, the next button will continue to push your progress forward it seems

**Comment by benalleng** [comment]
This should be closed as of our latest release through #1125 and #1126
